arch/risc-v/src/esp32c3/esp32c3_spiflash.c

```diff
@@ -0,0 +1,745 @@
+                         unsigned long arg)
+{
+  int ret = OK;
+  struct mtd_geometry_s *geo;
+
+  finfo("cmd: %d \n", cmd);
+
+  switch (cmd)
+    {
+      case MTDIOC_GEOMETRY:
+        {
+          geo = (struct mtd_geometry_s *)arg;
+          if (geo)
+            {
+              geo->blocksize    = SPI_FLASH_BLK_SIZE;
+              geo->erasesize    = SPI_FLASH_ERASE_SIZE;
+              geo->neraseblocks = SPI_FLASH_SIZE / SPI_FLASH_ERASE_SIZE;
+              ret               = OK;
+
+              finfo("blocksize: %" PRId32 " erasesize: %" PRId32 \
+                    " neraseblocks: %" PRId32 "\n",
+                    geo->blocksize, geo->erasesize, geo->neraseblocks);
+            }
+        }
+        break;
+
+      default:
+        ret = -ENOTTY;
+        break;
+    }
+
+  finfo("return %d\n", ret);
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32c3_spiflash_alloc_mtdpart
+ *
+ * Description:
+ *   Alloc ESP32-C3 SPI Flash MTD
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   ESP32-C3 SPI Flash MTD data pointer if success or NULL if fail
+ *
+ ****************************************************************************/
+
+FAR struct mtd_dev_s *esp32c3_spiflash_alloc_mtdpart(void)
+{
+  struct esp32c3_spiflash_s *priv = &g_esp32c3_spiflash;
+  esp32c3_spiflash_chip_t *chip = priv->chip;
+  FAR struct mtd_dev_s *mtd_part;
+  uint32_t blocks;
+  uint32_t startblock;
+  uint32_t size;
+
+  ASSERT((ESP32C3_MTD_OFFSET + ESP32C3_MTD_SIZE) <= chip->chip_size);
+  ASSERT((ESP32C3_MTD_OFFSET % chip->sector_size) == 0);
+  ASSERT((ESP32C3_MTD_SIZE % chip->sector_size) == 0);
+
+  finfo("ESP32 SPI Flash information:\n");
+  finfo("\tID = 0x%" PRIx32 "\n", chip->device_id);
+  finfo("\tStatus mask = 0x%" PRIx32 "\n", chip->status_mask);
+  finfo("\tChip size = %" PRId32 " KB\n", chip->chip_size / 1024);
+  finfo("\tPage size = %" PRId32 " B\n", chip->page_size);
+  finfo("\tSector size = %" PRId32 " KB\n", chip->sector_size / 1024);
+  finfo("\tBlock size = %" PRId32 " KB\n", chip->block_size / 1024);
+
+#if ESP32C3_MTD_SIZE == 0
+  size = chip->chip_size - ESP32C3_MTD_OFFSET;
+#else
+  size = ESP32C3_MTD_SIZE;
+#endif
+
+  finfo("\tMTD offset = 0x%x\n", ESP32C3_MTD_OFFSET);
+  finfo("\tMTD size = 0x%" PRIx32 "\n", size);
+
+  startblock = MTD_SIZE2BLK(priv, ESP32C3_MTD_OFFSET);
+  blocks = MTD_SIZE2BLK(priv, size);
+
+  mtd_part = mtd_partition(&priv->mtd, startblock, blocks);
+  if (!mtd_part)
+    {
+      ferr("ERROR: create MTD partition");
+      return NULL;
+    }
+
+  return mtd_part;
+}
+
+/****************************************************************************
+ * Name: esp32c3_spiflash_mtd
+ *
+ * Description:
+ *   Get ESP32-C3 SPI Flash MTD.
```

Comment:
```suggestion
 *   Get SPI Flash MTD.
```
I think it is redundant to state that this is related to ESP32-C3.